### PR TITLE
CON-754: Create query for "iam-no-hardware-mfa-root" in IAM

### PR DESCRIPTION
**Pull request by scvenema**
@martinianodl : Added ERB templating and triggers to what you sent me yesterday via slack

### CODE REVIEW

File: services/config.rb [hunk 1]
```diff
@@ -1124,6 +1124,45 @@
...
+      user_name
+      access_key_1_active
+      access_key_2_active
+      access_key_1_last_used_service
+      access_key_2_last_used_service
+      password_next_rotation
+      password_last_used
+      password_enabled
+    }
+    visualize(func: uid(no_mfa_root)) @filter(eq(tenant_id, "161866044093")) {
```
Comment (armaneous) [inline services/config.rb]: Probably don't want to leave `tenant_id` in there...

File: services/config.rb [hunk 2]
```diff
@@ -1124,6 +1124,45 @@
...
+      access_key_1_active
+      access_key_2_active
+      access_key_1_last_used_service
+      access_key_2_last_used_service
+      password_next_rotation
+      password_last_used
+      password_enabled
+      relates_to {
+        <%= default_predicates %>
+        relates_to @filter(NOT uid(no_mfa_root)){
```
Comment (armaneous) [inline services/config.rb]: This might need to be `NOT has(user)` because there could be shared objects between users. Unless that's okay?

Comment (scvenema) [reply services/config.rb]: You are talking about the 2nd `relates_to` filter?

File: services/config.rb [hunk 3]
```diff
@@ -1124,6 +1124,45 @@
...
+        <%= default_predicates %>
+        relates_to @filter(NOT has(user)) {
+          <%= default_predicates %>
+        }
+      }
+    }
+  }
+  QUERY
+  meta_rule_node_triggers({
+                              'user' => ['user_name', 'virtual_mfa_device']
```
Comment (armaneous) [inline services/config.rb]: I think a trigger needs to be added for the `mfa_active` property if I'm understanding the rule right. We want to trigger when that property changes between `true` and `false`.

**Pull request merged**